test/integration/suite_test.go

```diff
@@ -138,19 +139,21 @@ func TestMain(m *testing.M) {
 		ctx,
 		retry.Do(
 			func() error {
-				version, err := helpers.GetKongVersion(proxyAdminURL, consts.KongTestPassword)
+				kongVersion, err := helpers.ValidateMinimalSupportedKongVersion(proxyAdminURL, consts.KongTestPassword)
 				if err != nil {
 					return err
 				}
-				fmt.Printf("INFO: using Kong instance (version: %s) reachable at %s\n", version, proxyAdminURL)
+				fmt.Printf("INFO: using Kong instance (version: %q) reachable at %s\n", kongVersion, proxyAdminURL)
 				return nil
 			},
 			retry.OnRetry(
 				func(n uint, err error) {
 					fmt.Printf("WARNING: try to get Kong Gateway version attempt %d/10 - error: %s\n", n+1, err)
 				},
 			),
-			retry.LastErrorOnly(true),
+			retry.LastErrorOnly(true), retry.RetryIf(func(err error) bool {
+				return !errors.As(err, &helpers.TooOldKongGatewayError{})
```

Comment:
Does this work? 🤔 Typically `errors.As()` a "pointer to pointer" as in the example: https://pkg.go.dev/errors#example-As

Reply:
Yup it works as expected, see https://go.dev/play/p/J_27HpMWIi- because my error type has a value receiver and I return it as a value, it would be a different story for the pointer receiver and/or returning as pointer